packages/editor-ui/src/components/ParameterInputFull.vue

```diff
@@ -53,7 +59,74 @@ const focused = ref(false);
 const menuExpanded = ref(false);
 const forceShowExpression = ref(false);
 
+const fromAiOverride = {
+	overridePlaceholder: 'Defined automatically by the model',
+	icon: 'plus',
+	output: '={{ [marker] $fromAI([key], [type], [default], [description]) }}',
+	providers: {
+		marker: '/* n8n-auto-generated-override */',
+		key: (props: Props) => props.parameter.name, // todo - need to check these
+		type: (props: Props) => 'string', // todo - infer boolean, number and other matching types
+	},
+	extraProps: {
+		description: {
+			default: '',
+			placeholder: 'Description of how the model should define this value',
+		},
+	},
+};
+
+function buildValueFromOverride(
+	providedExtraProps: (typeof fromAiOverride)['extraProps'], // todo generic-ify
+	includeMarker: boolean,
+) {
+	const {
+		providers: { marker, key, type },
+		extraProps,
+	} = fromAiOverride;
+	return `={{ ${includeMarker ? marker : ''} $fromAI('${key(props)}', '${providedExtraProps.description.default ?? extraProps.description}', '${type(props)}') }}`;
+}
+
+function isOverrideValue(s: string) {
+	return s.startsWith('={{ /* n8n-auto-generated-override */ $fromAI(');
+}
+
+function parseOverrides(s: string) {
+	/*
+		Approaches:
+		- Smart and boring: Reuse packages/core/src/CreateNodeAsTool.ts - unclear where to move it
+		- Fun and dangerous: Use eval, see below
+	*/
+	if (!isOverrideValue(s)) return null;
+
+	const fromAI = (...args: unknown[]) => args;
+	const fns = `const $fromAI = ${fromAI}; const $fromAi = ${fromAI}; const $fromai = ${fromAI};`;
+	const cursor = '={{ /* n8n-auto-generated-override */ $fromAI('.length;
+	const end = '}}'.length;
+	// `eval?.` is an indirect evaluation, outside of local scope
+	const params: unknown[] = eval?.(`${fns} ${s.slice(cursor, -end)}`) ?? [];
```

Comment:
Can we avoid `eval` if possible? To avoid any security risks related to this.

Reply:
Wondering if this can be run within tournament since we already use it to safely resolve expressions

https://github.com/n8n-io/tournament?tab=readme-ov-file#usage